Backend/routes/customerOrderRoutes.js

```
@@ -161,23 +165,30 @@ router.put(
       if (order.status === "Cancelled") {
         return res.status(400).json({ error: "Order already cancelled" });
       }
 
       order.status = "Cancelled";
       await order.save();
 
-      const retailerOrder = await RetailerOrder.findOne({ shopId: order.shopId });
+      const retailerOrder = await RetailerOrder.findOne({ shopId: mongoose.Types.ObjectId(order.shopId) });
+
       if (retailerOrder) {
         const orderTotal = parseFloat(order.total.replace("$", ""));
```

Biome (1.9.4)

[error] Use Number.parseFloat instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseFloat instead.


(lint/style/useNumberNamespace)